src/mappings/spacewalk.ts

```diff
@@ -0,0 +1,184 @@
+    const vaultIdFlat = parseVaultId(vaultId)
+    const vault = await getOrCreateVault(
+        ctx,
+        vaultIdFlat,
+        vaultStellarPublicKey
+    )
+
+    const issueRequestStorage = await getVersionedStorage(
+        network,
+        ctx,
+        'issue',
+        'issueRequests'
+    )
+    const createdIssue = await issueRequestStorage.get(ctx.block, issueId)
+
+    const issueRequest = new IssueRequest({
+        id: issueId,
+        timestamp: new Date(ctx.block.timestamp ?? 0),
+        opentime: createdIssue.opentime,
+        period: createdIssue.period,
+        requester: hexToSs58(requester),
+        amount: amount,
+        vault: vault,
+        fee: fee,
+        griefingCollateral: griefingCollateral,
+        status: IssueRequestStatus.PENDING,
+    })
+    await ctx.store.save(issueRequest)
+}
+
+export async function handleIssueRequestExecuted(ctx: EventHandlerContext) {
+    const { args } = ctx.event
+    const { issueId, amount, fee } = args
+
+    let issueRequest = await ctx.store.get(IssueRequest, issueId)
+
+    if (issueRequest == null) {
+        throw new Error(
+            'Issue request MUST exists on the database when handling executed event'
+        )
+    }
+    // set property to completed, amount and fee could have also changed.
+    issueRequest.status = IssueRequestStatus.COMPLETED
+    issueRequest.amount = amount
+    issueRequest.fee = fee
+
+    await ctx.store.save(issueRequest)
+}
+
+export async function handleIssueRequestCancelled(ctx: EventHandlerContext) {
+    const { args } = ctx.event
+    const { issueId, griefingCollateral } = args
+
+    let issueRequest = await ctx.store.get(IssueRequest, issueId)
+
+    if (issueRequest == null) {
+        throw new Error(
+            'Issue request MUST exists on the database when handling cancel event'
+        )
+    }
+    // set property to completed. Griefing collateral slashed depends on when the requester cancelled, or if it expired.
+    issueRequest.status = IssueRequestStatus.CANCELLED
+    issueRequest.slashedCollateral = griefingCollateral
+
+    await ctx.store.save(issueRequest)
+}
+
+export async function handleIssueRequestAmountChanged(
+    ctx: EventHandlerContext
+) {
+    const { args } = ctx.event
+    const { issueId, confiscatedGriefingCollateral } = args
+
+    let issueRequest = await ctx.store.get(IssueRequest, issueId)
+
+    if (issueRequest == null) {
+        throw new Error(
+            'Issue request MUST exists on the database when handling cancel event'
+        )
+    }
+
+    // we update on this event the slashed collateral if the amount sent was not enough. The amount is updated on the executed event.
+    issueRequest.slashedCollateral = confiscatedGriefingCollateral
+    await ctx.store.save(issueRequest)
+}
+
+export async function handleRedeemRequest(ctx: EventHandlerContext) {
+    const { args } = ctx.event
+    const { redeemId, redeemer, amount, vaultId, fee, premium, transferFee } =
```

Comment:
Likewise, `stellarAddress and `asset` not interesting to us?

Reply:
I proposed we get it from the vault entity to avoid duplication. From `vaultIdFlat.wrapped`, if we ever need it.

Reply:
I can add the stellar address though, for the future.